### PR TITLE
add esx:spawnVehicle callback

### PR DESCRIPTION
**Pull request by Taavi-AU**
to spawn vehicles client side using a callback to server side spawn vehicles 👍

### CODE REVIEW

**Comment by Taavi-AU** [comment]
make changes if you want to but better than handling vehicle spawn callbacks in each resource so esx should have one in core which is easier.

**Comment by Taavi-AU** [comment]
fantik thats your **job**

File: [core]/es_extended/server/main.lua [hunk 1]
```diff
@@ -628,6 +628,22 @@ ESX.RegisterServerCallback('esx:getPlayerNames', function(source, cb, players)
...
 end)
 
+ESX.RegisterServerCallback("esx:spawnVehicle", function(source,cb,model,coords,heading,props,warp)
+  local ped = GetPlayerPed(source)
+  local vehicle = nil
+  props = props or {}
+  ESX.OneSync.SpawnVehicle(model, coords, heading, props, function(id)
+    vehicle = NetworkGetEntityFromNetworkId(id)
+    if warp then
+      while GetVehiclePedIsIn(ped) ~= vehicle do
```
Comment (Gellipapa) [inline [core]/es_extended/server/main.lua:638]: If will never be equal will run forever?

This should be an attempt for say 15 if it fails then break.

**Review by Gellipapa — CHANGES_REQUESTED** [review]

**Comment by Gellipapa** [comment]
@Taavi-AU Hi! Can you make requested changes?

File: [core]/es_extended/server/main.lua [hunk 2]
```diff
@@ -628,16 +628,21 @@ ESX.RegisterServerCallback('esx:getPlayerNames', function(source, cb, players)
   cb(players)
 end)
 
-ESX.RegisterServerCallback("esx:spawnVehicle", function(source,cb,model,coords,heading,props,warp)
+ESX.RegisterServerCallback("esx:spawnVehicle", function(source,cb,vehData)
   local ped = GetPlayerPed(source)
```
Comment (Gellipapa) [inline [core]/es_extended/server/main.lua:632]: Please handle if vehData is nil and vehData is table (we accept only table type) cb false and add log how "VehData is nil, please add vehData then parameters".

**Comment by Gellipapa** [comment]
@tony-stark-17 Hi! Thank you for fast work, please fix this request and i can merge good job, thank you :)

**Comment by TheFantomas** [comment]
> @tony-stark-17 Hi! Thank you for fast work, please fix this request and i can merge good job, thank you :)

+ Please handle formatting of this callback :D 4 spaces please

**Comment by Taavi-AU** [comment]
is that good, formatted, anything else?

**Comment by Taavi-AU** [comment]
example

```lua
ESX.TriggerServerCallback("esx:spawnVehicle", function(vehicleId)
	print(vehicleId)
end, {`ZENTORNO`, vector4(0.0, 0.0, 0.0, 180.0), {plate = "TEST123"}, true})
```

**Comment by Taavi-AU** [comment]
example

```lua
ESX.TriggerServerCallback("esx:spawnVehicle", function(vehicleId)
	print(vehicleId)
end, {model = `ADDER`, coords = vector4(0.0, 0.0, 0.0, 180.0), props = {plate = "TEST123"}, warp = true})
```

File: [core]/es_extended/server/main.lua [hunk 3]
```diff
@@ -628,6 +628,22 @@ ESX.RegisterServerCallback('esx:getPlayerNames', function(source, cb, players)
   cb(players)
 end)
 
+ESX.RegisterServerCallback("esx:spawnVehicle", function(source,cb,vehData)
+    local ped = GetPlayerPed(source)
+    ESX.OneSync.SpawnVehicle(vehData.model or `ADDER`, vehData.coords or GetEntityCoords(ped), vehData.coords.w or 0.0, vehData.props or {}, function(id)
+        local vehicle = NetworkGetEntityFromNetworkId(id)
```
Comment (Gellipapa) [inline [core]/es_extended/server/main.lua:634]: If vehData.wrap true  only use 

local vehicle = NetworkGetEntityFromNetworkId(id)

please put this if statement  thank you.

Comment (Gellipapa) [reply [core]/es_extended/server/main.lua:634]: @Taavi-AU Please do this change.

**Review by Gellipapa — APPROVED** [review]
It's okey, thank you. Looks cool.

**Pull request merged**